demos/time_dependent_rayleigh_benard/timedep-rayleigh-benard.py.rst

```diff
@@ -0,0 +1,186 @@
+Rayleigh-Benard Convection
+==========================
+
+Contributed by `Robert Kirby <https://sites.baylor.edu/robert_kirby/>`_
+and `Pablo Brubeck <https://www.maths.ox.ac.uk/people/pablo.brubeckmartinez/>`_.
+
+This problem involves a variable-temperature incompressible fluid.
+Variations in the fluid temperature are assumed to affect the momentum
+balance through a buoyant term (the Boussinesq approximation), leading
+to a Navier-Stokes equation with a nonlinear coupling to a
+convection-diffusion equation for temperature.
+
+We will set up the problem using Taylor-Hood elements for
+the Navier-Stokes part, and piecewise linear elements for the
+temperature. ::
+
+  from firedrake import *
+  from firedrake.pyplot import FunctionPlotter, tripcolor
+  import math
+  import matplotlib.pyplot as plt
+  from matplotlib.animation import FuncAnimation
+
+  try:
+      from irksome import Dt, RadauIIA, TimeStepper
+  except ImportError:
+      import sys
+      warning("Unable to import irksome.")
+      sys.exit(0)
```

Comment:
Rather than do this I would just include a bit like: "This tutorial assumes that you have Irksome installed as well as Firedrake. Instructions for installing Irksome can be found <here>."